csclient/params/params.go

```diff
@@ -258,6 +262,30 @@ type WhoAmIResponse struct {
 	Groups []string
 }
 
+// Resource holds the metadata about a Resource defined for a charm.
+type Resource struct {
+	Name        string
+	Type        ResourceType
```

Comment:
Why not just string?  This is a wire type.  Stricter bounds checking is facilitated through the helpers in charm/resource/type.go.